docs/guides.md

```diff
@@ -523,3 +523,307 @@ Now we enable autostart at the virtual machine level.
+
+This type of problem is very difficult to diagnose and correct.  It may be possible to resolve it using different BIOS or setup configurations in the host system or by updating BIOS or controller firmware.
+
+### More and Different
+
+So what if we don't have or don't want a system that's identical to the one we just built in these instructions?  Here are some of the possible and normal variations of software RAID under XCP-ng.
+
+#### Different Sized and Shaped RAID Arrays
+
+We might want to create a RAID array with more or fewer drives or a different RAID level.  Common types would be a two drive RAID 1 array or a 4 drive RAID 5, 6 or 10 array.  Those cases are very easy to accomodate by changing the parameters when building the RAID array, altering the level= or raid-devices= parameters and the list of drives on the mdadm --create command line.  The only other consideration is to make sure the drives used are accounted for in a DEVICE line in the mdadm.conf file.
+
+The number of drives in a specific level of RAID array can also affect the performance of the array.  A good rule of thumb for RAID 5 or 6 arrays is to have a number of drives that is a power of two (2, 4, 8, etc.) plus the number of extra drives where space is used for parity information, one drive for RAID 5 and two drives for RAID 6.  The RAID 5 array we created in the example system meets that recommendation by having 3 drives.  A 5 drive RAID5 array, or 4 or 6 drive RAID 6 arrays would as well.  A good rule of thumb for RAID 10 arrays is to have an even number of drives.  For RAID 10 in Linux an even number of drives is not a requirement as it is on other types of systems.  In addition, it may be possible to get better performance by creating a 2 drive RAID 10 array instead of a 2 drive RAID 1 array.
+
+#### Different Sized Drives
+
+We might need to create a RAID array where our drives are not identical and each drive has a different number of available blocks.  This might come up if we need to create a RAID array but have two of one type of drive and one of another such as two WD drives and one Seagate or two 1TB drives and one that is 1.5TB or 2TB.
+
+The easiest solution to creating a working RAID array in this situation is to partition the drives and create a RAID array using the partitions instead of using the entire drive.
+
+To do this, we get the sizes of the disks in the system by examining the /proc/partitions file.  Starting with the smallest of the disks to be used in the array, use gdisk or sgdisk to create a single partition of type fd00 (Linux RAID) using the maximum space available.  Examine and record the size of the partition created and save the changes.  Repeat the process with the remaining drives to be used except use the size of the partition created on the first drive instead of the maximum space available.
```

Comment:
```suggestion
To do this, we get the sizes of the disks in the system by examining the `/proc/partitions` file.  Starting with the smallest of the disks to be used in the array, use `gdisk` or `sgdisk` to create a single partition of type `fd00` (Linux RAID) using the maximum space available.  Examine and record the size of the partition created and save the changes.  Repeat the process with the remaining drives to be used except use the size of the partition created on the first drive instead of the maximum space available.
```